### PR TITLE
Connect 4!

### LINKED ISSUE #320: Add Connect 4 to the bot
## Description of proposed feature

Make a Connect 4 game for Codey. Some proposed commands for this are:

- `.connect4 [challenger]` - make a new game of Connect 4. You could either specify someone to challenge, or leave it "public" for anyone to join.

The game board would then be created. If the challenger was specified, they would need to join the game; if not, anyone could join the game. This could be done using some sort of button.

The two players could propose how much they are willing to bet on the game, and the winner takes all.

## How can the new feature be used?

Here's a possible screenshot of what the game could look like:
image

## Additional comments

If you have any suggestions for this, feel free to discuss in the comments of this issue.

### PR DESCRIPTION
**Pull request by SaurontheMighty**
## Summary of Changes
src/commands/games/connectFour.ts: Connect 4 command
src/commandDetails/games/connectFour.ts: Connect 4 command details
src/components/games/connectFour.ts: 
- Game Tracker
   - start a game
   - end a game
- Game: 
   - setStatus which checks for a win and sets status accordingly
   - getGameResponse
   - various helpers
- Connect 4 Game State
   - player ids
   - status
   - columns: array of ConnectFourColumn objects which in turn are arrays of ConnectFourGameSign objects which represent player tokens or pending if nothing there yet
   - winType: set to none at the start otherwise holds info about whether it was a horizontal/vertical/diagonal connect4
src/interaction-handlers/games/connectFour.ts: Interaction handlers for the buttons

## Motivation and Explanation
This PR adds Connect 4 to Codey. More games = better? This PR does not include bet or challenger. The user can simply play against codey who will pick a random column to put its token into.

## Related Issues
resolves #320 

## Steps to Reproduce
N/A

## Demonstration of Changes
image

### CODE REVIEW

File: src/commandDetails/games/connectFour.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+\`${container.botPrefix}connect4\`
+\`${container.botPrefix}connect 4 @user\``,
+
+  isCommandResponseEphemeral: false,
+  messageWhenExecutingCommand: 'Setting up your Connect 4 game...',
+  executeCommand: connectFourExecuteCommand,
+  afterMessageReply: connectFourAfterMessageReply,
+  options: [
+    {
+      name: 'challenger',
```
Comment (mcpenguin) [inline src/commandDetails/games/connectFour.ts:55]: Currently, connect4 only supports single player mode. I would remove this parameter for now until multiplayer is supported.

**Review by mcpenguin — CHANGES_REQUESTED** [review]
Small comment.

**Comment by SaurontheMighty** [comment]
> Seems like build is failing because of the linter.

Ah yeah one line I missed, fixed now.

**Review by mcpenguin — APPROVED** [review]
LGTM!

**Pull request merged**